### PR TITLE
Configuration parameter to switch key destination

### PR DESCRIPTION
**Pull request by KFilipek**
I've added configuration parameter for keys.

---
This change is [Reviewable](https://reviewable.io/reviews/pmem/redis/62)

### CODE REVIEW

**Comment by karpiop** [comment]
:lgtm:

---

Reviewed 6 of 6 files at r1.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io/reviews/pmem/redis/62#-:-LEPYLKe7PfNQAW2dLbu:bnfp4nl)*

File: src/config.c [hunk 1]
```diff
@@ -447,6 +447,8 @@ void loadServerConfigFromString(char *config) {
 			}
 			server.pm_file_size = size;
 			server.use_volatile = true;
+        } else if (!strcasecmp(argv[0],"keys-on-pm") && (argc == 2)) {
+            server.keys_on_pm = (!strcasecmp(argv[1], "on") ? true : false);
```
Comment (michalbiesek) [inline src/config.c]: consider validation of this parameter - now every value different than "on" is treated as "off".
If You look to other parameters it validate parameter properly value range, like for example methods: yesnotoi  or configEnumGetValue

File: src/server.c [hunk 2]
```diff
@@ -1456,6 +1463,7 @@ void initServerConfig(void) {
     server.pm_dir_path = NULL;
     server.pm_file_size = MEMKIND_PMEM_MIN_SIZE;
     server.use_volatile = true;
+    server.keys_on_pm = NULL;
```
Comment (michalbiesek) [inline src/server.c]: s/NULL/false

File: src/server.c [hunk 3]
```diff
@@ -584,7 +591,7 @@ dictType dbDictType = {
     NULL,                       /* key dup */
     NULL,                       /* val dup */
     dictSdsKeyCompare,          /* key compare */
-    dictSdsDestructorM,         /* key destructor */
+    dictSdsDestructorVar,         /* key destructor */
```
Comment (michalbiesek) [inline src/server.c]: cosmetics - but IMHO you could fix align in comment ( in simple word - remove two spaces)

File: src/server.c (hunk 2 above)
Comment (KFilipek) [reply src/server.c]: Good point

File: src/server.c (hunk 3 above)
Comment (KFilipek) [reply src/server.c]: Done

**Comment by michalbiesek** [comment]
:lgtm:

---

Reviewed 2 of 6 files at r1, 1 of 1 files at r2, 3 of 3 files at r3.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io/reviews/pmem/redis/62#-:-LETNawmATnpoUi1k2FB:bnfp4nl)*

**Pull request merged**